### PR TITLE
Make statsd endpoint configurable.

### PR DESCRIPTION
**Pull request by jk0**
This comes in handy if you are using a statsd proxy capable of forwarding to multiple vendors (DataDog, SignalFX, etc).

### CODE REVIEW

**Comment by jk0** [comment]
TODO: If `datadog_statsd_endpoint` is != 127.0.0.1, disable the dd-agent containers?

cc @bcwaldon does this seem reasonable?

**Comment by jk0** [comment]
@bcwaldon how's it look?

File: roles/kube-addons/tasks/main.yml [hunk 1]
```diff
@@ -6,9 +6,20 @@
...
 
+- name: deploy kube-datadog
+  include: deploy.yml
+  when: datadog_enabled
+  with_items:
+  - kube-datadog.yml
+
+- name: deploy datadog-agent
+  include: deploy.yml
+  when: (datadog_enabled and datadog_statsd_host == "127.0.0.1:8125")
```
Comment (gleeco) [inline roles/kube-addons/tasks/main.yml]: hardcoded not template var? 
either way, a comment here why this is might help de-baffle down the road.

**Comment by bcwaldon** [comment]
@jk0 two conceptual issues I can see:
1. Separating kube-datadog and its statsd-only datadog-agent means the kube-datadog addon is no longer going to be able to forward its logs in the case of `datadog_statsd_endpoint: 127.0.0.1:8125`. I believe that both containers need to stay in the same logical addon. What you can do here, however, is simply remove the datadog-agent container from the addon altogether if datadog_statsd_endpoint != 127.0.0.1:8125
2. The datadog-agent deployed onto each kubelet is actually there to collect host-level metrics. I believe this needs to remain even if datadog_statsd_endpoint != 127.0.0.1:8125. You could, however, forward these metrics off to the custom statsd endpoint.

**Comment by bcwaldon** [comment]
@jk0 👍 after you squash it all down

**Pull request merged**